src/lib/data/storage/PersistentStorage.cpp

```diff
@@ -1024,35 +1024,45 @@ std::shared_ptr<Graph> PersistentStorage::getGraphForAll() const
 {
 	TRACE();
 
-	std::vector<Id> tokenIds;
-
-	m_sqliteIndexStorage.forEach<StorageNode>([&](StorageNode&& node) {
-		bool showNode = true;
-		if (m_symbolDefinitionKinds.size())
-		{
-			auto it = m_symbolDefinitionKinds.find(node.id);
-			showNode = (it != m_symbolDefinitionKinds.end() && it->second == DEFINITION_EXPLICIT);
-		}
-
-		if (showNode &&
-			(NodeType(NodeType::intToType(node.type)).isPackage() ||
-			 !m_hierarchyCache.isChildOfVisibleNodeOrInvisible(node.id)))
-		{
-			tokenIds.push_back(node.id);
-		}
-	});
-
-	for (const auto& p: m_fileNodeIndexed)
+	std::shared_ptr<Graph> graph = std::make_shared<Graph>();
+	if (m_symbolDefinitionKinds.size()) {
+		m_sqliteIndexStorage.forEach<StorageNode>([&](StorageNode&& storageNode) {
+			const NodeType type(NodeType::intToType(storageNode.type));
+			if (type.isFile())
+			{
+				addFileNodeToGraph(storageNode, graph.get());
```

Comment:
You still need to check whether a file is indexed in `m_fileNodeIndexed`. Otherwise all files are shown in the overview, not only the indexed ones (in TicTacToe sample now 261 instead of 9).